### PR TITLE
exui-1006-avoid hidden field deletion fr case flag

### PR DESCRIPTION
**Pull request by RiteshHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EXUI-998
https://tools.hmcts.net/jira/browse/EXUI-1000
https://tools.hmcts.net/jira/browse/EXUI-1006

### Change description ###
Complex type not showing data
Case Flags stopped working

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by richardhenryash — DISMISSED** [review]

**Review by andywilkinshmcts — APPROVED** [review]

**Review by richardhenryash — DISMISSED** [review]

File: projects/ccd-case-ui-toolkit/src/lib/shared/pipes/complex/ccd-read-fields-filter.pipe.spec.ts [hunk 1]
```diff
@@ -278,4 +323,25 @@ describe('ReadFieldsFilterPipe', () => {
     expect(RESULT[1].hidden).toEqual(true);
     expect(RESULT[2].hidden).toEqual(true);
   });
+  it('it shoulld evaluate showcondition and set the hidden property of field to true when value doesnt match within complex field even Formgroup passed with idPrefix passed as empty string', () => {
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/pipes/complex/ccd-read-fields-filter.pipe.spec.ts:326]: @RiteshHMCTS - please remove the second "it" and correct typo to `should`. Also, "doesnt" should be `doesn\'t` (backslash escape needed). These corrections are also needed on 5 unit tests above this one

File: projects/ccd-case-ui-toolkit/src/lib/shared/pipes/complex/ccd-read-fields-filter.pipe.spec.ts [hunk 2]
```diff
@@ -278,4 +323,25 @@ describe('ReadFieldsFilterPipe', () => {
...
+    complexCaseField2.value = {
+      caseAccepted: 'Yes',
+      dateAccepted: '10/01/2023'
+    };
+    const RESULT: CaseField[] = pipe.transform(complexCaseField2, false, undefined, true, FORM_GROUP.controls['data'], undefined, '');
+    expect(RESULT.length).toEqual(2);
+    expect(RESULT[0].hidden).toEqual(false);
+    expect(RESULT[1].hidden).toEqual(true);
+  });
+  it('it shoulld evaluate showcondition and set the hidden property of field to false when value doesnt match within complex field even Formgroup passed with idPrefix passed as empty string', () => {
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/pipes/complex/ccd-read-fields-filter.pipe.spec.ts:336]: @RiteshHMCTS - please remove the second "it" and correct typo to `should`. Also, "doesnt" should be `doesn\'t` (backslash escape needed)

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@RiteshHMCTS - a few minor changes suggested; everything else is OK

File: projects/ccd-case-ui-toolkit/src/lib/shared/pipes/complex/ccd-read-fields-filter.pipe.spec.ts (hunk 1 above)
Comment (RiteshHMCTS) [reply projects/ccd-case-ui-toolkit/src/lib/shared/pipes/complex/ccd-read-fields-filter.pipe.spec.ts:326]: @Daniel-Lam thanks for your review, addressed above comments

File: projects/ccd-case-ui-toolkit/src/lib/shared/pipes/complex/ccd-read-fields-filter.pipe.spec.ts (hunk 2 above)
Comment (RiteshHMCTS) [reply projects/ccd-case-ui-toolkit/src/lib/shared/pipes/complex/ccd-read-fields-filter.pipe.spec.ts:336]: @Daniel-Lam addressed above comments

**Review by richardhenryash — APPROVED** [review]

**Review by Daniel-Lam — APPROVED** [review]

**Pull request merged**